bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/slider/AjaxNumericBootstrapSlider.java

```diff
@@ -0,0 +1,25 @@
+package de.agilecoders.wicket.extensions.slider;
+
+import de.agilecoders.wicket.extensions.slider.util.*;
+import org.apache.wicket.model.IModel;
+
+/**
+ * AjaxNumericBootstrapSlider
+ */
+public class AjaxNumericBootstrapSlider<T extends Number> extends AjaxBootstrapSlider<INumericValue<T>, T> {
+
+    public AjaxNumericBootstrapSlider(String id, IModel<T> model) {
+        super(id, new NumericModel(model), (Class<INumericValue<T>>)getSliderClass(model.getObject().getClass()));
```

Comment:
Just to avoid passing the class... The model object should be non null anyway: either slider will not know where to place handles. It is needed to decide which SliderValue (Long, Integer, Double) will be used.